azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/NetworkInterface.java

```diff
@@ -70,7 +70,7 @@
      *
      * @return public ip resource id
      */
-    String publicIpAddressId();
+    String primaryPublicIpAddressId();
```

Comment:
since we have publicIpAddress(), do we really need publicIpAddressId()?
if the perf of fetching the PIP is an issue, and this is meant as a quick way to get at the id without the server roundtrip, then we could make the PIP itself "refreshable" - the idea we chatted abnout recently - that we could return an "empty shell" of a PIP, only with the id, but not yet hydrated with all the other data. And hydrate it only when the first property on it is actually getting accessed.
But this perf improvement could also be donein the future if needed - in any case, we could I think just eliminate this duplication with publicIpAddressId.